assets/js/swedbank-pay-design-guide-theme.js

```diff
@@ -20,54 +20,95 @@
 (function () {
+    const pathHash = window.location.pathname + window.location.hash;
+    const originPath = window.location.origin + window.location.pathname
+    const activeGroup = [...document.querySelectorAll(".nav-group")]
+    .filter(group => group.querySelector(`[href="${pathHash}"], [href^="${originPath}#"]`))[0];
+    const activeSubgroup = [...document.querySelectorAll(".nav-subgroup")].filter(group => group.querySelector(`[href="${pathHash}"], [href^="${originPath}#"]`))
+    
+    activeGroup.classList.add("active")
+    activeSubgroup.map(subgroup => subgroup.classList.add("active"))
+
     var headings = document.querySelectorAll("h2");
-    var tocLinks = document.querySelectorAll("nav.doc-toc ul li a");
+    var tocLinks = document.querySelectorAll("nav.sidebar-nav .nav-subgroup.active .nav-leaf");
+
+    if (tocLinks.length == 0) {
+        var tocLinks = document.querySelectorAll("nav.sidebar-nav .nav-group.active .nav-leaf");
+    }
 
     var getPosition = function (el) {
         if (el) {
             var bodyRect = document.body.getBoundingClientRect();
             var elemRect = el.getBoundingClientRect();
-
+            
             return elemRect.top - bodyRect.top;
         }
-
+        
         return null;
     };
 
-    window.addEventListener("scroll", function () {
-        var buffer = document.body.clientHeight * 0.1;
-        var currentPos = window.pageYOffset + buffer;
-
-        // TODO: Probably a stupid way to compute "how far left can we scroll until
-        //       we reach the bottom of the page", but it seems to work.
-        var scrollDistanceFromBottom = document.documentElement.scrollHeight
-            - document.documentElement.scrollTop
-            - document.body.clientHeight
-            - buffer;
+    var setActiveLeafScroll = function () {
+        if (tocLinks.length > 0) {
+            var activeLeaf = document.querySelector("nav.sidebar-nav .nav-leaf.active");
+            var buffer = document.body.clientHeight * 0.1;
+            var currentPos = window.pageYOffset + buffer;
+    
+            // TODO: Probably a stupid way to compute "how far left can we scroll until
+            //       we reach the bottom of the page", but it seems to work.
+            var scrollDistanceFromBottom = document.documentElement.scrollHeight
+                - document.documentElement.scrollTop
+                - document.body.clientHeight
+                - buffer;
+                
+            const scrollNumber = [...headings].filter(heading => getPosition(heading) <= currentPos).length - 1;
 
-        if (scrollDistanceFromBottom <= 0) {
-            for (var link of tocLinks) {
-                link.parentElement.classList.remove("active");
+            activeLeaf && activeLeaf.classList.remove("active");
+            
+            
+            if (scrollDistanceFromBottom > 0) {
+                scrollNumber >= 0 && tocLinks[scrollNumber].classList.add("active");

```

Comment:
Codacy found an issue: ['tocLinks' used outside of binding context.](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-design-guide-jekyll-theme/pullRequest?prid=5659803)